### PR TITLE
Add true and false as PseudoTypes

### PR DESCRIPTION
**Pull request by mvriel**
In this change, I have made more allowance for identifying types as
PseudoTypes and to ensure they are identified as both their underlying
and pseudo-type.

Using instanceof, you can identify whether it is the PseudoType, a
pseudotype, but also as the underlying type.

### CODE REVIEW

File: src/PseudoTypes/False_.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+use phpDocumentor\Reflection\PseudoType;
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\Types\Boolean;
+
+/**
+ * Value Object representing the PseudoType 'False', which is a Boolean type.
+ *
+ * @psalm-immutable
+ */
+final class False_ extends Boolean implements PseudoType
```
Comment (jaapio) [inline src/PseudoTypes/False_.php:26]: Moving this class to another namespace isn't possible :-( becaus of BC.

File: src/PseudoTypes/True_.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use phpDocumentor\Reflection\PseudoType;
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\Types\Boolean;
+
+/**
+ * Value Object representing the PseudoType 'False', which is a Boolean type.
+ *
+ * @psalm-immutable
+ */
+final class True_ extends Boolean implements PseudoType
```
Comment (jaapio) [inline src/PseudoTypes/True_.php:26]: Moving this class to another namespace isn't possible :-( becaus of BC.

**Review by jaapio — CHANGES_REQUESTED** [review]

File: src/PseudoTypes/False_.php (hunk 1 above)
Comment (mvriel) [reply src/PseudoTypes/False_.php:26]: As discussed on Slack: I'll add a class_alias to the old name. The new namespace helps clarify

File: src/PseudoTypes/True_.php (hunk 2 above)
Comment (mvriel) [reply src/PseudoTypes/True_.php:26]: As discussed on Slack: I'll add a class_alias to the old name. The new namespace helps clarify

File: src/PseudoTypes/False_.php (hunk 1 above)
Comment (jaapio) [reply src/PseudoTypes/False_.php:26]: It might be good to introduce a deprecation warning on those aliases :-)

**Comment by mvriel** [comment]
The BC break mentioned in the build is mitigated by using class_alias in the False_ and True_ classes. The tool is apparently unable to detect that and I even added a test to check the BC

File: src/PseudoTypes/False_.php (hunk 1 above)
Comment (mvriel) [reply src/PseudoTypes/False_.php:26]: I have been unable to add a deprecation warning; at least: I don't know how I would accomplish that

**Review by jaapio — APPROVED** [review]

**Pull request merged**